shuup/admin/modules/settings/__init__.py

```diff
@@ -0,0 +1,125 @@
+}
+
+
+class ViewSettings(object):
+    key = "view_configuration_%s_%s"
+    column_spec = []
+
+    def __init__(self, model, default_columns, *args, **kwargs):
+
+        if isinstance(model, six.string_types) and model == settings.AUTH_USER_MODEL:
+            model = apps.get_model(settings.AUTH_USER_MODEL)
+
+        self.model = model
+        self.default_columns = default_columns
+        self.ensure_initial_columns(default_columns)
+        self.column_spec = self._build_settings_columns()
+        self.columns = [column for column in self.column_spec if self.get_config(column.id)]
+
+    def ensure_initial_columns(self, default_columns):
+        if not self.view_configured():
+            # generate the default stuff
+            for default_column in default_columns:
+                self.set_config(default_column.id, 1)
+            self.set_config("saved", True)
+
+    def get_config(self, value):
+        val = configuration.get(None, self.get_settings_key(value), None)
+        return val
+
+    def set_config(self, key_value, value, use_key=False):
+        key = self.get_settings_key(key_value) if not use_key else key_value
+        return configuration.set(None, key, value)
+
+    def get_settings_key(self, value):
+        return self.key % (self.model.__name__.lower(), value)
+
+    def view_configured(self):
+        return self.get_config("saved")
+
+    def _valid_field(self, field_identifier):
+        return not len([field for field in INVALID_FIELDS if field in field_identifier])
+
+    def _build_settings_columns(self):
+        columns = []
+
+        if hasattr(self.model, "_parler_meta"):
+            for field_id, field in self.model._parler_meta.get_fields_with_model():
+                column = self._get_translated_column(field_id, field)
+                columns.append(column)
+
+        for field in self.model._meta.local_fields:
+            column = self._get_column(field)
+            if column:
+                columns.append(column)
+
+        return columns
+
+    def _get_translated_column(self, field_id, field):
+        field = getattr(self.model, field_id).field.meta.model._meta.get_field(field_id)
+        field_name = field.verbose_name.title()
+        filter_config = TextFilter(
+            filter_field="translations__%s" % field_id,
+            placeholder=field_name
+        )
+        column = Column(
+            field_id,
+            field_name,
+            sort_field="translations__%s" % field_id,
+            display=field_id,
+            filter_config=filter_config
+        )
+        return self.handle_special_column(field, column)[0]
+
+    def handle_special_column(self, field, column):
+        # reverse handlers for convenience
+        is_special = False
+        for original_column in self.default_columns:
+            if column.id == original_column.id:
+                is_special = True
+                if getattr(original_column, "sortable"):
```

Comment:
I'm actually thinking that should we even limit these values and just overwrite anything that the original one defines?